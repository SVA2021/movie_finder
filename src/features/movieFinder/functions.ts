export const normalizeDateString = (date: number): string => {
	let result;
	(date < 10) ? result = '0' + String(date) : result = String(date);
	return result;
}

export const convertNumToTime = (num: number | null): string => {
	let result;
	if (num === null) return '';
	let hours = Math.floor(num % 60);
	let minutes = num % 60;

	result = (hours === 0) ? `${minutes}мин ` : `${hours}ч ${minutes}мин`;
	return result;
}

export const convertRatingToColor = (num: number| null): string => {
	if (num === null) return '';
	if (num <= 0 || num > 10) return 'white';
	if (num >= 8) return 'green';
	if (num >= 6) return 'orange';
	return 'red';
}