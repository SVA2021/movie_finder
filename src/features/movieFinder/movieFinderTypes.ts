//********** BASIC CONSTANTS *************************/

export const PRODUCTION_STATUS = ['FILMING', 'PRE_PRODUCTION', 'COMPLETED', 'ANNOUNCED', 'UNKNOWN', 'POST_PRODUCTION'] as const;
export const FILM_TYPE = ['FILM', 'VIDEO', 'TV_SERIES', 'MINI_SERIES', 'TV_SHOW'] as const;
export const TOP_FILMS_TYPE = ['TOP_250_BEST_FILMS', 'TOP_100_POPULAR_FILMS', 'TOP_AWAIT_FILMS'] as const;

//********** BASIC TYPES *************************/

export type TCountry = {country: string}
export type TGenre = {genre: string}
export type TProductionStatus = typeof PRODUCTION_STATUS[number];
export type TFilmType = typeof FILM_TYPE[number];
export type TTopList = typeof TOP_FILMS_TYPE[number];
export type TUser = {user: string, password: string} | null;
export type TStatus = 'idle' | 'loading' | 'failed';

//********** ITEM TYPES *************************/

export interface TSmallCard {
  id: number
  nameRu: string | null
  nameEn: string | null
  year: string | number | null
  filmLength: string | number | null
  rating: string | null
  posterUrlPreview: string
}

export interface Film {
  filmId: number
  nameRu: string | null
  nameEn: string | null
  year: string | null
  filmLength: string | null
  countries: TCountry[]
  genres: TGenre[]
  rating: string | null
  ratingVoteCount: number | null
  posterUrl: string
  posterUrlPreview: string
}

export interface TTopFilm {
  filmId: number
  nameRu: string | null
  nameEn: string | null
  year: string | null
  filmLength: string | null
  countries: TCountry[]
  genres: TGenre[]
  rating: string | null
  ratingVoteCount: number | null
  posterUrl: string
  posterUrlPreview: string
  ratingChange?: null | any
}

export interface TFullCard {
  kinopoiskId: number
  imdbId: string | null
  nameRu: string | null
  nameEn: string | null
  nameOriginal: string | null
  posterUrl: string
  posterUrlPreview: string
  coverUrl: string | null
  logoUrl: string | null
  reviewsCount: number
  ratingGoodReview: number | null
  ratingGoodReviewVoteCount: number | null
  ratingKinopoisk: number | null
  ratingKinopoiskVoteCount: number | null
  ratingImdb: number | null
  ratingImdbVoteCount: number | null
  ratingFilmCritics: number | null
  ratingFilmCriticsVoteCount: number | null
  ratingAwait: number | null
  ratingAwaitCount: number | null
  ratingRfCritics: number | null
  ratingRfCriticsVoteCount: number | null
  webUrl: string
  year: number | null
  filmLength: number | null
  slogan: string | null
  description: string | null
  shortDescription: string | null
  editorAnnotation: string | null
  isTicketsAvailable: boolean
  productionStatus: TProductionStatus | null
  type: TFilmType
  ratingMpaa: string | null
  ratingAgeLimits: string | null
  countries: TCountry[]
  genres: TGenre[]
  startYear?: number | null
  endYear?: number | null
  serial?: boolean | null
  shortFilm?: boolean | null
  completed?: boolean | null
  hasImax?: boolean | null
  has3D?: boolean | null
  lastSync?: string
}

export type TSearchItem = Pick<TFullCard, 'kinopoiskId' | 'imdbId' | 'nameRu' | 'nameEn' | 'nameOriginal'
  | 'countries' | 'genres' | 'ratingKinopoisk' | 'ratingImdb' | 'year' | 'type' | 'posterUrl' | 'posterUrlPreview'>;

//********** RESPONSE TYPES *************************/

export interface TSearchResponse {
  total: number
  totalPages: number
  items: TSearchItem[]
}

export interface TSearchResponseData {
  total: number | null
  totalPages: number | null
  currentPage: number
  keyword: string | null
  items: {[key: number]: TSearchItem[]}
}

export interface TTopResponse {
  pagesCount: number
  films: TTopFilm[]
  type: TTopList
  page: number
}

export interface TTopData {
  pagesCount: number
  page: number
  data: TSmallCard[]
}

export interface THomePageCurrent {
  type: TTopList
  page: number
}

export interface TMovieDetails {
  movie: TFullCard | null
  similars: TSimilarsData | null
  videos: TVideosRes | null
  images: TImagesRes | null
  awards: TAwardsRes | null
  facts: TFactsRes | null
}

// export type TMovieExtraType = 'similars' | 'videos' | 'images' | 'awards' | 'facts'
export type TMovieExtraType = 'videos' | 'images' | 'awards' | 'facts'
export type TVideoItemType = 'YOUTUBE' | 'KINOPOISK_WIDGET' | 'UNKNOWN'

export interface TMovieSimilarsRes {
  total: number
  items: Film[]
}

export interface TSimilarsData {
  total: number
  data: TSmallCard[]
}

export interface TVideoItem {
  url: string
  name: string
  site: TVideoItemType
}

export interface TImageItem {
  imageUrl: string
  previewUrl: string
}

export interface TVideosRes {
  total: number
  items: TVideoItem[]
}

export interface TImagesRes {
  total: number
  totalPages: number
  items: TImageItem[]
}

export interface TAwardsItem {
  name: string
  win: boolean
  imageUrl: string
  nominationName: string
  year: number
}

export interface TAwardsRes {
  total: number
  items: TAwardsItem[]
}

export interface TFactsItem {
  text: string
  spoiler: boolean
  type: 'FACT' | 'BLOOPER'
}

export interface TFactsRes {
  total: number
  items: TFactsItem[]
}

export interface TMovieExtraProps {
  id: number
}

export interface TSearchRequest {
  keyword: string | null
  page: number
  // extended: TSearchExtended | null
}

export const ORDERS = ['RATING', 'NUM_VOTE', 'YEAR'] as const;
export type TOrder = typeof ORDERS[number];
// export type TOrder = 'RATING' | 'NUM_VOTE' | 'YEAR'
export type TFilterGenre = {id: number, genre: string}
export type TFilterCountry = {id: number, country: string}

export interface TFiltersResponse {
  genres: TFilterGenre[]
  countries: TFilterCountry[]
}

export interface TSearchForm {
  countries: number | 'default'
  genres: number | 'default'
  order: TOrder
  rating: [number, number]
  year: [number, number]
}

export interface TSearchExtended {
  countries: number[] | null
  genres: number[] | null
  order: TOrder
  ratingFrom: number
  ratingTo: number
  yearFrom: number
  yearTo: number
}

export interface TSearch {
  search: TSearchRequest
  extended: TSearchForm | null
}