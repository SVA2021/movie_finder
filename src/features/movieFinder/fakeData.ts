import { TFullCard, TSmallCard } from "./movieFinderTypes";

// export const filmDataShort: TMovieShort = {
// 	filmId: 263531,
// 	nameRu: 'FakeNameRu',
// 	nameEn: 'FakeNameEng',
// 	type: 'FILM',
// 	year: '2000',
// 	description: 'FakeDescription',
// 	filmLength: 100,
// 	countries: [{ country: 'FakeCountry' }],
// 	genres: [{ genre: 'FakeGenre' }],
// 	rating: '85%',
// 	ratingVoteCount: 1000,
// 	posterUrl: 'https://kinopoiskapiunofficial.tech/images/posters/kp/301.jpg',
// 	posterUrlPreview: 'https://kinopoiskapiunofficial.tech/images/posters/kp_small/301.jpg',
// }

// export const fakeSmallCard: TSmallCard = {
// 	filmId: 263531,
// 	nameRu: 'FakeNameRu',
// 	nameEn: 'FakeNameEng',
// 	year: 2000,
// 	filmLength: 100,
// 	countries: [{ country: 'FakeCountry' }],
// 	genres: [{ genre: 'FakeGenre' }],
// 	rating: 7.9,
// 	ratingVoteCount: 1000,
// 	posterUrl: 'https://kinopoiskapiunofficial.tech/images/posters/kp/301.jpg',
// 	posterUrlPreview: 'https://kinopoiskapiunofficial.tech/images/posters/kp_small/301.jpg',
// }

// export const fakeSmallCard: TSmallCard = {
// 	"id": 395787,
// 	"nameRu": "Семь жизней",
// 	"nameEn": "Seven Pounds",
// 	"year": "2008",
// 	"filmLength": "01:58",
// 	"countries": [
// 		{
// 			"country": "США"
// 		}
// 	],
// 	"genres": [
// 		{
// 			"genre": "драма"
// 		},
// 		{
// 			"genre": "мелодрама"
// 		}
// 	],
// 	"rating": "8.2",
// 	"ratingVoteCount": 344640,
// 	"posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/395787.jpg",
// 	"posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/395787.jpg",
// 	"ratingChange": null
// }

export const fakeFullCard: TFullCard = {
	"kinopoiskId": 555,
	"imdbId": "tt0118715",
	"nameRu": "Большой Лебовски",
	"nameEn": null,
	"nameOriginal": "The Big Lebowski",
	"posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/555.jpg",
	"posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/555.jpg",
	"coverUrl": "https://avatars.mds.yandex.net/get-ott/239697/2a000001612cac1f659b8bf59d99a62d073c/orig",
	"logoUrl": null,
	"reviewsCount": 142,
	"ratingGoodReview": 93.3,
	"ratingGoodReviewVoteCount": 128,
	"ratingKinopoisk": 7.8,
	"ratingKinopoiskVoteCount": 242932,
	"ratingImdb": 8.1,
	"ratingImdbVoteCount": 792300,
	"ratingFilmCritics": 7.5,
	"ratingFilmCriticsVoteCount": 109,
	"ratingAwait": null,
	"ratingAwaitCount": 0,
	"ratingRfCritics": null,
	"ratingRfCriticsVoteCount": 2,
	"webUrl": "https://www.kinopoisk.ru/film/555/",
	"year": 1998,
	"filmLength": 117,
	"slogan": "It takes guys as simple as the Dude and Walter to make a story this complicated... and they'd really rather be bowling",
	"description": "Лос-Анджелес, 1991 год, война в Персидском заливе. Главный герой по прозвищу Чувак считает себя совершенно счастливым человеком. Его жизнь составляют игра в боулинг и выпивка. Но внезапно его счастье нарушается, гангстеры по ошибке принимают его за миллионера-однофамильца, требуют деньги, о которых он ничего не подозревает, и, ко всему прочему, похищают жену миллионера, будучи уверенными, что «муж» выплатит за нее любую сумму.",
	"shortDescription": "Тихого пофигиста по ошибке принимают за богача. Криминальная комедия братьев Коэн, которая разошлась на цитаты",
	"editorAnnotation": null,
	"isTicketsAvailable": false,
	"productionStatus": null,
	"type": "FILM",
	"ratingMpaa": "r",
	"ratingAgeLimits": "age18",
	"countries": [
		{
			"country": "США"
		},
		{
			"country": "Великобритания"
		}
	],
	"genres": [
		{
			"genre": "криминал"
		},
		{
			"genre": "комедия"
		}
	],
	"startYear": null,
	"endYear": null,
	"serial": false,
	"shortFilm": false,
	"completed": false,
	"hasImax": false,
	"has3D": false,
	"lastSync": "2022-06-05T12:58:15.228012"
}

export const fakeResponseByKeyword = {
  "total": 76,
  "totalPages": 4,
  "items": [
    {
      "kinopoiskId": 278522,
      "imdbId": "tt0903624",
      "nameRu": "Хоббит: Нежданное путешествие",
      "nameEn": null,
      "nameOriginal": "The Hobbit: An Unexpected Journey",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Новая Зеландия"
        }
      ],
      "genres": [
        {
          "genre": "приключения"
        },
        {
          "genre": "боевик"
        },
        {
          "genre": "фэнтези"
        }
      ],
      "ratingKinopoisk": 8.1,
      "ratingImdb": 7.8,
      "year": 2012,
      "type": "FILM",
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/278522.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/278522.jpg"
    },
  ]
};

export const TOP2 = {
  "pagesCount": 13,
  "films": [
    {
      "filmId": 395787,
      "nameRu": "Семь жизней",
      "nameEn": "Seven Pounds",
      "year": "2008",
      "filmLength": "01:58",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "мелодрама"
        }
      ],
      "rating": "8.2",
      "ratingVoteCount": 351371,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/395787.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/395787.jpg",
      "ratingChange": null
    },
    {
      "filmId": 458,
      "nameRu": "Корпорация монстров",
      "nameEn": "Monsters, Inc.",
      "year": "2001",
      "filmLength": "01:32",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "приключения"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "мультфильм"
        },
        {
          "genre": "семейный"
        }
      ],
      "rating": "8.1",
      "ratingVoteCount": 409339,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/458.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/458.jpg",
      "ratingChange": null
    },
    {
      "filmId": 714248,
      "nameRu": "Песнь моря",
      "nameEn": "Song of the Sea",
      "year": "2014",
      "filmLength": "01:33",
      "countries": [
        {
          "country": "Франция"
        },
        {
          "country": "Дания"
        },
        {
          "country": "Ирландия"
        },
        {
          "country": "Бельгия"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "приключения"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "мультфильм"
        },
        {
          "genre": "семейный"
        }
      ],
      "rating": "8.1",
      "ratingVoteCount": 81444,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/714248.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/714248.jpg",
      "ratingChange": null
    },
    {
      "filmId": 530,
      "nameRu": "Игры разума",
      "nameEn": "A Beautiful Mind",
      "year": "2001",
      "filmLength": "02:15",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "мелодрама"
        },
        {
          "genre": "биография"
        }
      ],
      "rating": "8.5",
      "ratingVoteCount": 448759,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/530.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/530.jpg",
      "ratingChange": null
    },
    {
      "filmId": 327,
      "nameRu": "Крестный отец 2",
      "nameEn": "The Godfather: Part II",
      "year": "1974",
      "filmLength": "03:22",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "8.5",
      "ratingVoteCount": 182923,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/327.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/327.jpg",
      "ratingChange": null
    },
    {
      "filmId": 399,
      "nameRu": "Храброе сердце",
      "nameEn": "Braveheart",
      "year": "1995",
      "filmLength": "02:58",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "биография"
        },
        {
          "genre": "военный"
        },
        {
          "genre": "история"
        }
      ],
      "rating": "8.3",
      "ratingVoteCount": 253676,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/399.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/399.jpg",
      "ratingChange": null
    },
    {
      "filmId": 104938,
      "nameRu": "В погоне за счастьем",
      "nameEn": "The Pursuit of Happyness",
      "year": "2006",
      "filmLength": "01:57",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "биография"
        }
      ],
      "rating": "8.3",
      "ratingVoteCount": 313870,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/104938.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/104938.jpg",
      "ratingChange": null
    },
    {
      "filmId": 456,
      "nameRu": "Унесённые ветром",
      "nameEn": "Gone with the Wind",
      "year": "1939",
      "filmLength": "03:42",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "мелодрама"
        },
        {
          "genre": "военный"
        },
        {
          "genre": "история"
        }
      ],
      "rating": "8.4",
      "ratingVoteCount": 172328,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/456.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/456.jpg",
      "ratingChange": null
    },
    {
      "filmId": 77269,
      "nameRu": "Шерлок Холмс и доктор Ватсон: Знакомство",
      "nameEn": null,
      "year": "1979",
      "filmLength": "01:08",
      "countries": [
        {
          "country": "СССР"
        }
      ],
      "genres": [
        {
          "genre": "криминал"
        },
        {
          "genre": "детектив"
        }
      ],
      "rating": "8.6",
      "ratingVoteCount": 144506,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/77269.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/77269.jpg",
      "ratingChange": null
    },
    {
      "filmId": 450213,
      "nameRu": "Невидимая сторона",
      "nameEn": "The Blind Side",
      "year": "2009",
      "filmLength": "02:09",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "биография"
        },
        {
          "genre": "спорт"
        }
      ],
      "rating": "8.2",
      "ratingVoteCount": 231733,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/450213.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/450213.jpg",
      "ratingChange": null
    },
    {
      "filmId": 280172,
      "nameRu": "Как приручить дракона",
      "nameEn": "How to Train Your Dragon",
      "year": "2010",
      "filmLength": "01:38",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "приключения"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "мультфильм"
        },
        {
          "genre": "семейный"
        }
      ],
      "rating": "8.2",
      "ratingVoteCount": 492918,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/280172.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/280172.jpg",
      "ratingChange": null
    },
    {
      "filmId": 7103,
      "nameRu": "Назад в будущее 3",
      "nameEn": "Back to the Future Part III",
      "year": "1990",
      "filmLength": "01:58",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "приключения"
        },
        {
          "genre": "вестерн"
        },
        {
          "genre": "комедия"
        }
      ],
      "rating": "8.2",
      "ratingVoteCount": 310734,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/7103.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/7103.jpg",
      "ratingChange": null
    },
    {
      "filmId": 42172,
      "nameRu": "Тайна третьей планеты",
      "nameEn": null,
      "year": "1981",
      "filmLength": "00:50",
      "countries": [
        {
          "country": "СССР"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "приключения"
        },
        {
          "genre": "мультфильм"
        },
        {
          "genre": "семейный"
        }
      ],
      "rating": "8.1",
      "ratingVoteCount": 119051,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/42172.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/42172.jpg",
      "ratingChange": null
    },
    {
      "filmId": 9691,
      "nameRu": "Бесславные ублюдки",
      "nameEn": "Inglourious Basterds",
      "year": "2009",
      "filmLength": "02:33",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Германия"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "боевик"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "военный"
        }
      ],
      "rating": "7.9",
      "ratingVoteCount": 424593,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/9691.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/9691.jpg",
      "ratingChange": null
    },
    {
      "filmId": 5277,
      "nameRu": "Мулан",
      "nameEn": "Mulan",
      "year": "1998",
      "filmLength": "01:24",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "приключения"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "военный"
        },
        {
          "genre": "мультфильм"
        },
        {
          "genre": "семейный"
        },
        {
          "genre": "мюзикл"
        }
      ],
      "rating": "8.1",
      "ratingVoteCount": 242690,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/5277.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/5277.jpg",
      "ratingChange": null
    },
    {
      "filmId": 278522,
      "nameRu": "Хоббит: Нежданное путешествие",
      "nameEn": "The Hobbit: An Unexpected Journey",
      "year": "2012",
      "filmLength": "02:49",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Новая Зеландия"
        }
      ],
      "genres": [
        {
          "genre": "приключения"
        },
        {
          "genre": "боевик"
        },
        {
          "genre": "фэнтези"
        }
      ],
      "rating": "8.1",
      "ratingVoteCount": 455782,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/278522.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/278522.jpg",
      "ratingChange": null
    },
    {
      "filmId": 5273,
      "nameRu": "Шрэк 2",
      "nameEn": "Shrek 2",
      "year": "2004",
      "filmLength": "01:33",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "мелодрама"
        },
        {
          "genre": "приключения"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "мультфильм"
        },
        {
          "genre": "семейный"
        }
      ],
      "rating": "7.7",
      "ratingVoteCount": 257786,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/5273.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/5273.jpg",
      "ratingChange": null
    },
    {
      "filmId": 441,
      "nameRu": "Принцесса Мононоке",
      "nameEn": "Mononoke-hime",
      "year": "1997",
      "filmLength": "02:14",
      "countries": [
        {
          "country": "Япония"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "приключения"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "мультфильм"
        },
        {
          "genre": "аниме"
        }
      ],
      "rating": "8.2",
      "ratingVoteCount": 137308,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/441.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/441.jpg",
      "ratingChange": null
    },
    {
      "filmId": 102124,
      "nameRu": "Пираты Карибского моря: На краю света",
      "nameEn": "Pirates of the Caribbean: At World's End",
      "year": "2007",
      "filmLength": "02:49",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "приключения"
        },
        {
          "genre": "боевик"
        },
        {
          "genre": "фэнтези"
        }
      ],
      "rating": "8.0",
      "ratingVoteCount": 440467,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/102124.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/102124.jpg",
      "ratingChange": null
    },
    {
      "filmId": 470191,
      "nameRu": "Невероятный мир глазами Энцо",
      "nameEn": "The Art of Racing in the Rain",
      "year": "2019",
      "filmLength": "01:49",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "мелодрама"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "семейный"
        },
        {
          "genre": "спорт"
        }
      ],
      "rating": "8.2",
      "ratingVoteCount": 94545,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/470191.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/470191.jpg",
      "ratingChange": null
    }
  ]
};