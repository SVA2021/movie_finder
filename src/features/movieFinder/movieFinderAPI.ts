import axios from "axios";
import { API_OFFICIAL } from "../../private/key";
import { TTopList } from "./movieFinderTypes";

export const BASE_URL = 'https://kinopoiskapiunofficial.tech/';
const FILMS = 'api/v2.2/films/';
const TOP = 'top/';

const movieFinderInstance = axios.create({
	baseURL: BASE_URL,
	headers: {
		'X-API-KEY': API_OFFICIAL,
		'Content-Type': 'application/json',
	},
});

export const movieFinderAPI = {

	getTopList(topList: TTopList = 'TOP_250_BEST_FILMS', page: number = 1) {
		let query = `${FILMS}${TOP}?page=${page}`;
		return movieFinderInstance.get(query)
			.then(response => {
				console.log(response);
				return response;
			})
			.catch(e => {
				console.log(e);
				return e;
			});
	},

	getMovieData(id: number = 1) {
		let query = `${FILMS}${id}`;
		return movieFinderInstance.get(query)
			.then(response => {
				console.log(response);
				return response;
			})
			.catch(e => {
				console.log(e);
				return e;
			});
	},

	getMoviesByKeyword(keyword: string | undefined, page: number = 1) {
		let query = `${FILMS}?keyword=${keyword}&page=${page}`;
		return movieFinderInstance.get(query)
			.then(response => {
				console.log(response);
				return response;
			})
			.catch(e => {
				console.log(e);
				return e;
			});
	},

	getSeries(page: number = 1) {
		let query = `${FILMS}?type=TV_SERIES&page=${page}`;
		return movieFinderInstance.get(query)
			.then(response => {
				console.log(response);
				return response;
			})
			.catch(e => {
				console.log(e);
				return e;
			});
	},
}
