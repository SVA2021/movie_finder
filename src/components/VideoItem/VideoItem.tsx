import {Box, Link, Typography} from '@mui/material';
import React, {FC} from 'react';
import {TVideoItem, TVideoItemType} from '../../features/movieFinder/movieFinderTypes';
import YouTubeIcon from '@mui/icons-material/YouTube';
import LiveTvIcon from '@mui/icons-material/LiveTv';
import LocalMoviesIcon from '@mui/icons-material/LocalMovies';

function getVideoItem(type: TVideoItemType) {
    switch (type) {
        case 'KINOPOISK_WIDGET':
            return LocalMoviesIcon;
        case 'YOUTUBE':
            return YouTubeIcon;
        case 'UNKNOWN':
            return LiveTvIcon;
    }
}

interface VideoItemProps {
    item: TVideoItem
}

export const VideoItem: FC<VideoItemProps> = ({item}) => {

    const VideoIcon = getVideoItem(item.site);

    return (
        <Box p={2} height={'200px'} border={'1px solid'} borderColor={'secondary.main'} borderRadius={4} >
            <Link target={'_blank'} rel={'noopener noreferrer'}
                href={item.url}
                height={'100%'}
                component={'a'}
                title={'Смотреть в новом окне'}
                sx={{
                    display: 'flex',
                    flexDirection: 'column',
                    alignItems: 'center',
                    justifyContent: 'space-evenly',
                }}
            >
                <VideoIcon fontSize={'large'} />
                <Typography
                    variant={'body1'}
                    component={'p'}
                    color={'text.secondary'}
                    textAlign={'center'}
                >
                    {item.name}
                </Typography>
            </Link>
        </Box>
    );
};