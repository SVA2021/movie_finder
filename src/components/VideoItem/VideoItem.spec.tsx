import {render, screen} from "@testing-library/react";
import {TVideoItem} from "../../features/movieFinder/movieFinderTypes";
import {VideoItem} from "./VideoItem";

const fakeItem: TVideoItem = {
  url: 'fakeUrl',
  name: "fakeName",
  site: "YOUTUBE"
}

describe('render VideoItem', () => {

  const setUp = () => render(<VideoItem item={fakeItem} />);

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('videoitem')).toMatchSnapshot();
  });

  it('check text', () => {
    setUp();
    expect(screen.getByText(/fakeName/i)).toBeInTheDocument();
  });

  it('check link href', () => {
    setUp();
    expect(screen.getByRole('link')).toHaveAttribute('href', 'fakeUrl')
  });

})