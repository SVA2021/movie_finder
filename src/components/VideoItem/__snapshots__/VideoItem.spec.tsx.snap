// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render VideoItem snapshot test 1`] = `
<div
  class="MuiBox-root css-fncluw"
  data-testid="videoitem"
>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-16t8mm7-MuiTypography-root-MuiLink-root"
    href="fakeUrl"
    rel="noopener noreferrer"
    target="_blank"
    title="Смотреть в новом окне"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
      data-testid="YouTubeIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z"
      />
    </svg>
    <p
      class="MuiTypography-root MuiTypography-body1 css-19f7wsr-MuiTypography-root"
    >
      fakeName
    </p>
  </a>
</div>
`;
